sparkle/output/configuration_output.py

```diff
@@ -0,0 +1,256 @@
+                                        cr: ConfigurationResults) -> dict:
+        """Transform ConfigurationResults to dictionary format."""
+        return {
+            "performance": {
+                "configured_metrics": cr.performance["default_metrics"],
+                "default_metrics": cr.performance["configured_metrics"],
+            },
+            "configured_results": {
+                "solver": cr.configured_results.solver.name,
+                "configuration": cr.configured_results.configuration,
+                "instance_set": cr.configured_results.instance_set.name,
+                "result_header": cr.configured_results.result_header,
+                "result_values": cr.configured_results.result_vals,
+            },
+            "default_results": {
+                "solver": cr.default_results.solver.name,
+                "configuration": cr.default_results.configuration,
+                "instance_set": cr.default_results.instance_set.name,
+                "result_header": cr.default_results.result_header,
+                "result_values": cr.default_results.result_vals,
+            }
+        }
+
+    def serialize_scenario(self: ConfigurationOutput,
+                           cs: ConfigurationScenario) -> dict:
+        """Transform ConfigurationScenario to dictionary format."""
+        return {
+            "number_of_runs": cs.number_of_runs,
+            "solver_calls": cs.solver_calls,
+            "cpu_time": cs.cpu_time,
+            "wallclock_time": cs.wallclock_time,
+            "cutoff_time": cs.cutoff_time,
+            "cutoff_length": cs.cutoff_length,
+            "sparkle_objective": cs.sparkle_objective.name,
+            "use_features": cs.use_features,
+            "configurator_target": cs.configurator_target,
+            "feature_data": cs.feature_data,
+        }
+
+    def write_output(self: ConfigurationOutput) -> None:
+        """Write data into a JSON file."""
+        output_data = {
+            "solver": self.solver.name if self.solver else None,
+            "configurator": (
+                str(self.configurator.executable_path) if self.configurator else None
+            ),
+            "best_configuration": self.best_config,
+            "configurations": self.configurations,
+            "scenario": self.serialize_scenario(self.configurator.scenario)
+            if self.configurator.scenario else None,
+            "training_set": (
+                self.serialize_configuration_results(self.training)
+                if self.training else None
+            ),
+            "test_set": (
+                self.serialize_configuration_results(self.test) if self.test else None
+            ),
+        }
+
+        self.output.parent.mkdir(parents=True, exist_ok=True)
+        with self.output.open("w") as f:
+            json.dump(output_data, f, indent=4)
+        print("Analysis of configuration can be found here: ", self.output)
+
+
+if __name__ == "__main__":
```

Comment:
I believe you told me this is to be removed, if so I have nothing to comment. Tip: You could put this into a PyTest to make your life easier. I can show you how if you want me to